sdk/inc/azure/core/az_mqtt5_rpc_server_hfsm.h

```diff
@@ -262,42 +166,42 @@ typedef struct
 /**
  * @brief Starts the MQTT5 RPC Server.
  *
- * @param[in] client The #az_mqtt5_rpc_server to start.
+ * @param[in] client The #az_mqtt5_rpc_server_hfsm to start.
  *
  * @return An #az_result value indicating the result of the operation.
  */
-AZ_NODISCARD az_result az_mqtt5_rpc_server_register(az_mqtt5_rpc_server* client);
+AZ_NODISCARD az_result az_mqtt5_rpc_server_register(az_mqtt5_rpc_server_hfsm* client);
 
 /**
  * @brief Initializes an MQTT5 RPC Server.
  *
- * @param[out] client The #az_mqtt5_rpc_server to initialize.
+ * @param[out] client The #az_mqtt5_rpc_server_hfsm to initialize.
  * @param[in] rpc_server_codec The #az_mqtt5_rpc_server_codec to initialize and use within the RPC
  * Server.
  * @param[in] connection The #az_mqtt5_connection to use for the RPC Server.
  * @param[in] property_bag The application allocated #az_mqtt5_property_bag to use for the
  * RPC Server.
  * @param[in] subscription_topic The application allocated #az_span to use for the subscription
  * topic.
+ * @param[in] service_group_id The service group id to use for the subscription topic. May be
+ * AZ_SPAN_EMPTY if not used in the subscription topic.
  * @param[in] model_id The model id to use for the subscription topic. May be AZ_SPAN_EMPTY if not
  * used in the subscription topic.
  * @param[in] client_id The client id to use for the subscription topic. May be AZ_SPAN_EMPTY if not
  * used in the subscription topic.
- * @param[in] command_name The command name to use for the subscription topic or AZ_SPAN_EMPTY to
- * have this rpc server handle all commands for this topic.
  * @param[in] options Any #az_mqtt5_rpc_server_codec_options to use for the RPC Server.
  *
  * @return An #az_result value indicating the result of the operation.
  */
 AZ_NODISCARD az_result az_mqtt5_rpc_server_init(
-    az_mqtt5_rpc_server* client,
```

Comment:
Is there a need to call these `_hfsm`? Can we have:
- `rpc_server_codec` for the codec.
- `rpc_server` the stateful implementation?